### PR TITLE
Clean-up after nix-shell implementation

### PR DESCRIPTION
**Pull request by zuzuleinen**
closes https://github.com/benchkram/bob/issues/252

List of environment variables kept for `nix-shell --pure`:
https://github.com/NixOS/nix/blob/c0559a1d6092b94c7ce46a065d72fc74e4db51dd/src/nix-build/nix-build.cc#L100

To properly test output of Nix build dependencies run `nix-collect-garbage -d` before build to cleanup the store

### CODE REVIEW

**Comment by Equanox** [comment]
It's too easy for a user to think, "hey nothing is happening". My suggestion is to print a ellipsis every 5sec. You can then print the actual result in the next line. Ideally you can show how long it took to build/download ( i leave this up to you if it's easily possible)?
```
nodejs-16_x: ............
nodejs-16_x: /nix/store/dvzrdz86i15bmjyy869mi7h2bcgl05az-nodejs-16.15.0  took 28s
```
image

File: bob/global/env.go [hunk 1]
```diff
@@ -1,4 +0,0 @@
-package global
-
-// EnvWhitelist keep the whitelisted env variables when hermetic mode is on
-var EnvWhitelist = []string{"HOME", "XDG_CACHE_HOME"}
```
Comment (Equanox) [inline bob/global/env.go:4]: It looks to me that XDG_CACHE_HOME is not kept

image

Comment (zuzuleinen) [reply bob/global/env.go:4]: I removed our implementation of whitelisting env variables and relied on `nix-shell --pure` whitelisting strategy:

https://github.com/NixOS/nix/blob/c0559a1d6092b94c7ce46a065d72fc74e4db51dd/src/nix-build/nix-build.cc#L100

If there is a reason to keep XDG_CACHE_HOME I can revert and add it back

Comment (zuzuleinen) [reply bob/global/env.go:4]: Marking as resolved:

"Yes i think there was a reason for it
I remember go uses this to reuse the cache on the users home 
Please keep it"

**Comment by zuzuleinen** [comment]
https://github.com/benchkram/bob/pull/256 has to be merged before because it contains some fixes on output as well. Having a branch with https://github.com/benchkram/bob/pull/256 and current PR together will show correct output

**Review by Equanox — APPROVED** [review]

**Pull request merged**